lib/widgets/stream_colors.dart

```diff
@@ -0,0 +1,252 @@
+import 'dart:ui';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_color_models/flutter_color_models.dart';
+
+import '../api/model/model.dart';
+import 'color.dart';
+
+/// A caching factory for [StreamColorSwatch]es, from [subscription.color] bases.
```

Comment:
You give it a `[subscription.color]`, and it returns a `[StreamColorSwatch]` for it. If there's one cached, it returns that, otherwise it creates one, caches it, and returns it.

(I guess "creates one" is just what I was trying to get at with "factory".)